src/hevm/src/EVM/UnitTest.hs

```diff
@@ -548,14 +549,19 @@ word32Bytes :: Word32 -> ByteString
 word32Bytes x = BS.pack [byteAt x (3 - i) | i <- [0..3]]
 
 setupCall :: TestVMParams -> Text -> AbiValue -> EVM ()
-setupCall TestVMParams{..} sig args  = do
+setupCall TestVMParams{..} sig args = do
   resetState
-  use (env . contracts) >>= assign (tx . txReversion)
   assign (tx . isCreate) False
   loadContract testAddress
-  assign (state . calldata) $ (ConcreteBuffer $ abiMethod sig args, literal . num . BS.length $ abiMethod sig args)
+  assign (state . calldata) (ConcreteBuffer $ abiMethod sig args, literal . num . BS.length $ abiMethod sig args)
   assign (state . caller) (litAddr testCaller)
   assign (state . gas) (w256 testGasCall)
+  origin' <- fromMaybe (initialContract (RuntimeCode mempty)) <$> use (env . contracts . at testOrigin)
+  let originBal = view balance origin'
+  when (originBal <= (w256 testGasprice) * (w256 testGasCall)) $ error "insufficient balance for gas cost"
+  modifying (env . contracts) (setupTx testOrigin testCoinbase (w256 testGasprice) (w256 testGasCall))
+  vm <- get
+  put $ initTx vm $ view (env . contracts) vm
```

Comment:
Why do we need `setupTx` and `initTx`, they seem like they are doing very similar things?

Reply:
hm, yes can maybe be unified